blog/views.py

```
@@ -0,0 +1,36 @@
+from django.http import HttpRequest, HttpResponse
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.shortcuts import render, redirect
+from django.urls import reverse_lazy
+from django.views import generic
+
+
+from .models import Post, Commentary
+from .forms import CommentaryForm
+
+
+class PostListView(generic.ListView):
+    model = Post
+    queryset = Post.objects.prefetch_related("owner").order_by("-created_time")
+    paginate_by = 5
+    template_name = "blog/index.html"
+
+
+class PostDetailView(generic.DetailView):
+    model = Post
+
+    def post(self, request: HttpRequest, *args, **kwargs):
+        self.object = self.get_object()
+        form = CommentaryForm(request.POST)
+        if form.is_valid() and request.user.is_authenticated:
```

The POST handler saves the comment only when `form.is_valid() and request.user.is_authenticated`. This means an anonymous POST will be silently ignored (and the form won't be marked invalid). The requirement says: *If an anonymous user tries to create a new comment, the form must be invalid.* Consider checking authentication first and, if not authenticated, call `form.add_error(None, 'You must be logged in to comment')` (or otherwise invalidate the form) before evaluating/saving.